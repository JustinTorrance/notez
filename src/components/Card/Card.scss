.Card {
  border: 1px solid grey;
  min-width: 200px;
  margin: 1.5rem;
  border-radius: 4px;
  box-shadow: 1px 2px 6px grey;
}

.note-title {
  padding-left: 1rem; 
}

.note-list {
  display: flex;
  justify-content: space-between;
  list-style: none;
  padding: 1rem;
}

.note-delete-button {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 1rem;
}

