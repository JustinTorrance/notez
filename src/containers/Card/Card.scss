.Card {
  border: 1px solid #d3d3d3;
  min-width: 200px;
  max-width: 500px;
  margin: 1.5rem;
  border-radius: 4px;
  box-shadow: 1px 2px 6px grey;
}

.note-title {
  padding-left: 1rem; 
}

.note-list {
  display: flex;
  justify-content: space-between;
  list-style: none;
  padding: 1rem;
}

.note-delete-button-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 1rem;
}

.note-delete-button {
  border-radius: 4px;
  &:hover {
    background-color: hotpink;
    cursor: pointer;
  }
}

.checkbox {
  margin-right: .5rem;
  &:hover {
    cursor: pointer;
  }
}

.list-item-delete-button {
  border-radius: 4px;
  &:hover {
    background-color: hotpink;
    cursor: pointer;
  }
}