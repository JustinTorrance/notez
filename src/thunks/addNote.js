export const addNote = (url, newNote) => {
  return async (dispatch) => {
    const options = {
      method: "POST",
      header: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(newNote)
    }
    try {
      const response = await fetch(url, options)
      const data = await response.json()
      console.log('data', data)
    } catch(error) {
      console.log(error)
    }
  }
}

