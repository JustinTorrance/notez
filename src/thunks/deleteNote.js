import { fetchErrored, isLoading, deleteNoteAction } from '../actions'

export const deleteNote = (url, id) => {
  return async (dispatch) => {
    const options = {
      method: "DELETE", 
    }
    try {
      dispatch(isLoading(true));
      const response = await fetch(url, options);
      if (!response.ok) {
        throw new Error(response.statusText)
      } 
      dispatch(isLoading(false));
      dispatch(deleteNoteAction(id));
    } catch(error) {
      dispatch(fetchErrored(error.message))
    }
  }
}