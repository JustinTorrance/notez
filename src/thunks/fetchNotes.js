import { getNotes, isLoading, fetchErrored } from '../actions';

export const fetchNotes = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const notes = await response.json();
      dispatch(getNotes(notes));
      dispatch(isLoading(false));
    } catch(error) {
      dispatch(fetchErrored(error.message));
    }
  }
}