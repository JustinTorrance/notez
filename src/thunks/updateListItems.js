import { fetchErrored } from '../actions'

export const updateListItems = (url, updatedNote) => {
  return async (dispatch) => {
    const options = {
      method: "PUT",
      body: JSON.stringify(updatedNote),
      headers: {
        "Content-Type": "application/json"
      }
    }
    try {
      const response = await fetch(url, options)
      if (!response.ok) {
        throw new Error(response.statusText)
      } 
    } catch(error) {
      dispatch(fetchErrored(error.message))
    }
  }
}

 