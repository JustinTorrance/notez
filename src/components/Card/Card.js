import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteNote } from '../../thunks/deleteNote';
import { updateListItems } from '../../thunks/updateListItems';
import PropTypes from 'prop-types';

class Card extends Component {

  separateCheckedItems = () => {
    return this.props.listItems.filter(listItem => {
      return listItem.completed;
    });
  }

  separateUncheckedItems = () => {
    return this.props.listItems.filter(listItem => {
      return !listItem.completed;
    });
  }

  displayNoteText = (sortedListItems) => {
    return sortedListItems.map(listItem => {
      return (
        <div className='note-list'>
          <li className='list-items' style={{ textDecorationLine: listItem.completed? 'line-through' : 'none' }}>
            <input
              className='checkbox'
              type='checkbox'
              onChange={this.checkedListItem}
              id={listItem.id}
            />
            {listItem.text}
          </li>
          <button onClick={(e) => this.deleteListItem(e)} id={listItem.id} className='list-item-delete-button'>x</button>
        </div>
      )
    })
  };

  deleteListItem = (e) => {
    console.log('target', e.target.id)
    const listItems = this.props.listItems.filter(item => (item.id != e.target.id))
    const { title, id } = this.props
    const revisedNote = { title, id, listItems }
    const url = `http://localhost:3001/api/v1/notes/${id}`;
    this.props.updateListItems(url, revisedNote)
    
  }

  //line 34 invoke deleteListITem
  // filter over this.props.listItems
// return this.props.listItems != e.target.id
//updateListItems(url, newObject)
// invoke retrieveNotes 

  checkedListItem = (e) => {
    const foundListItem = this.props.listItems.find(item => {
      return e.target.id == item.id
    });
    this.updateNote(foundListItem.id);
  };

  updateNote = (itemId) => {
    const { id, title, listItems } = this.props;
    const url = `http://localhost:3001/api/v1/notes/${id}`;
    this.props.listItems.map(listItem => {
      if (listItem.id === itemId) {
        return listItem.completed = !listItem.completed;
      } else {
        return listItem
      }
    });
    const object = {id, title, listItems}
    this.props.updateListItems(url, object);
  };

  deleteNote = (e) => {
    const { value } = e.target;
    const url = `http://localhost:3001/api/v1/notes/${value}`;
    this.props.deleteNote(url, value)
  };

  render() {
    const { title, id, listItems } = this.props
    return (
      <div className='Card'>
        <h3 className='note-title'>{title}</h3>
        {this.displayNoteText(this.separateUncheckedItems())}
        {this.displayNoteText(this.separateCheckedItems())}
        <div className='note-delete-button-wrapper'>
          <button className='note-delete-button' onClick={this.deleteNote} value={id}>DELETE NOTE</button>
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  notes: PropTypes.array.isRequired,
  deleteNote: PropTypes.func.isRequired,
  updateListItems: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  listItems: PropTypes.array.isRequired
};

export const mapStateToProps = (state) => ({
  notes: state.notes
});

export const mapDispatchToProps = (dispatch) => ({
  deleteNote: (url, id) => dispatch(deleteNote(url, id)),
  updateListItems: (url, updatedListItem) => dispatch(updateListItems(url, updatedListItem))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);
