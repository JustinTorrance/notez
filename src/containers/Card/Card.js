import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteNote } from '../../thunks/deleteNote';
import { updateListItems } from '../../thunks/updateListItems';
import PropTypes from 'prop-types';
import { toggleCompleted, deleteListItem } from '../../actions';

export class Card extends Component {

  separateCheckedItems = () => {
    return this.props.listItems.filter(listItem => {
      return listItem.completed;
    });
  }

  separateUncheckedItems = () => {
    return this.props.listItems.filter(listItem => {
      return !listItem.completed;
    });
  }

  displayNoteText = (sortedListItems) => {
    return sortedListItems.map(listItem => {
      return (
        <div className='note-list' key={listItem.id}>
          <li className='list-items' style={{ textDecorationLine: listItem.completed? 'line-through' : 'none' }}>
            <input
              className='checkbox'
              type='checkbox'
              onChange={this.checkedListItem}
              id={listItem.id}
            />
            {listItem.text}
          </li>
          <button onClick={(e) => this.deleteListItem(e)} id={listItem.id} className='list-item-delete-button'>x</button>
        </div>
      )
    })
  };

  deleteListItem = (e) => {
    const listItems = this.props.listItems.filter(item => (item.id != e.target.id));
    const { title, id } = this.props;
    const revisedNote = { title, id, listItems };
    const url = `http://localhost:3001/api/v1/notes/${id}`;
    this.props.deleteListItem(e.target.id);
    this.props.updateListItems(url, revisedNote);
  };

  checkedListItem = (e) => {
    const foundListItem = this.props.listItems.find(item => {
      return e.target.id == item.id
    });
    this.updateNote(foundListItem.id);
    this.props.toggleCompleted(e.target.id);
  };

  updateNote = (itemId) => {
    const { id, title, listItems } = this.props;
    const url = `http://localhost:3001/api/v1/notes/${id}`;
    this.props.listItems.map(listItem => {
      if (listItem.id == itemId) {
        return listItem.completed = !listItem.completed;
      } else {
        return listItem
      }
    });
    const object = {id, title, listItems}
    this.props.updateListItems(url, object);
  };

  deleteNote = (e) => {
    const { value } = e.target;
    const url = `http://localhost:3001/api/v1/notes/${value}`;
    this.props.deleteNote(url, value)
  };

  render() {
    const { title, id } = this.props
    return (
      <div className='Card'>
        <h3 className='note-title'>{title}</h3>
        {this.displayNoteText(this.separateUncheckedItems())}
        {this.displayNoteText(this.separateCheckedItems())}
        <div className='note-delete-button-wrapper'>
          <button className='note-delete-button' onClick={this.deleteNote} value={id}>DELETE NOTE</button>
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  notes: PropTypes.array.isRequired,
  deleteNote: PropTypes.func.isRequired,
  updateListItems: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  listItems: PropTypes.array.isRequired
};

export const mapStateToProps = (state) => ({
  notes: state.notes
});

export const mapDispatchToProps = (dispatch) => ({
  deleteNote: (url, id) => dispatch(deleteNote(url, id)),
  updateListItems: (url, updatedListItem) => dispatch(updateListItems(url, updatedListItem)),
  toggleCompleted: (id) => dispatch(toggleCompleted(id)),
  deleteListItem: (id) => dispatch(deleteListItem(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);
