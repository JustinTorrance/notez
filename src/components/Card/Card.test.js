import React from 'react';
import { Card } from '../Card/Card';
import { shallow } from 'enzyme';
import { deleteNote } from '../../thunks/deleteNote';


describe('deleteNote', () => {
  let wrapper;
  let mockUrl;
  let mockNotes;
  let mockId;
  let mockTitle;
  let mockFunction;

  beforeEach(() => {
    mockUrl = 'www.somenotes.com';
    mockId = '1';
    mockTitle = "mockTitle"
    mockNotes = [
      { id: '1', title: 'fakeTitle', listItems: [{id: 1, text: 'faketext'}] },
      { id: '2', title: 'fakeTitle2', listItems: [{id: 2, text: 'faketext2'}] }
    ];
    wrapper = shallow( 
      <Card 
        key={mockId}
        deleteNote={deleteNote} 
        id={mockId} 
        listItems={mockNotes} 
        title={mockTitle}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the deleteNote prop with url and value', () => {
    let e.target = 
    wrapper.instance().deleteNote();
    expect(mockFunction).toHaveBeenCalled()
    
  });
});