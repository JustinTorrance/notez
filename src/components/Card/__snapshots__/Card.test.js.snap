// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteNote should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    deleteNote={[Function]}
    id="1"
    listItems={
      Array [
        Object {
          "id": "1",
          "listItems": Array [
            Object {
              "id": 1,
              "text": "faketext",
            },
          ],
          "title": "fakeTitle",
        },
        Object {
          "id": "2",
          "listItems": Array [
            Object {
              "id": 2,
              "text": "faketext2",
            },
          ],
          "title": "fakeTitle2",
        },
      ]
    }
    title="mockTitle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          mockTitle
        </h3>,
        Array [
          <div>
            <li />
            <button>
              DELETE LIST-ITEM
            </button>
          </div>,
          <div>
            <li />
            <button>
              DELETE LIST-ITEM
            </button>
          </div>,
        ],
        <button
          onClick={[Function]}
          value="1"
        >
          DELETE NOTE
        </button>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mockTitle",
        },
        "ref": null,
        "rendered": "mockTitle",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li />,
            <button>
              DELETE LIST-ITEM
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DELETE LIST-ITEM",
            },
            "ref": null,
            "rendered": "DELETE LIST-ITEM",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li />,
            <button>
              DELETE LIST-ITEM
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DELETE LIST-ITEM",
            },
            "ref": null,
            "rendered": "DELETE LIST-ITEM",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DELETE NOTE",
          "onClick": [Function],
          "value": "1",
        },
        "ref": null,
        "rendered": "DELETE NOTE",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            mockTitle
          </h3>,
          Array [
            <div>
              <li />
              <button>
                DELETE LIST-ITEM
              </button>
            </div>,
            <div>
              <li />
              <button>
                DELETE LIST-ITEM
              </button>
            </div>,
          ],
          <button
            onClick={[Function]}
            value="1"
          >
            DELETE NOTE
          </button>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mockTitle",
          },
          "ref": null,
          "rendered": "mockTitle",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li />,
              <button>
                DELETE LIST-ITEM
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DELETE LIST-ITEM",
              },
              "ref": null,
              "rendered": "DELETE LIST-ITEM",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li />,
              <button>
                DELETE LIST-ITEM
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DELETE LIST-ITEM",
              },
              "ref": null,
              "rendered": "DELETE LIST-ITEM",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DELETE NOTE",
            "onClick": [Function],
            "value": "1",
          },
          "ref": null,
          "rendered": "DELETE NOTE",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
